pinot-common/src/main/java/org/apache/pinot/common/data/FieldSpec.java

```diff
@@ -439,6 +415,28 @@ public int size() {
           throw new IllegalStateException("Cannot get number of bytes for: " + this);
       }
     }
+
+    /**
+     * Converts the given string value to the data type.
+     */
+    public Object convert(String value) {
```

Comment:
probably not part of this PR but why is this method part of the FieldSpec?

Reply:
This is part of the DataType, which is inside FieldSpec.